Cabal/Distribution/Types/LocalBuildInfo.hs

```diff
@@ -301,3 +308,15 @@ externalPackageDeps lbi =
     -- True if this dependency is an internal one (depends on the library
     -- defined in the same package).
     internal ipkgid = any ((==ipkgid) . componentUnitId) (Graph.toList (componentGraph lbi))
+
+withProfLib :: LocalBuildInfo -> Bool
+withProfLib = isJust . libProfiling
```

Comment:
Well, it does look like ide-backend is the ONLY client, so I think it's OK if you decide to rename.